.github/CONTRIBUTING.md

```diff
@@ -1,32 +1,122 @@
+    and finally https://github.com/checkstyle/checkstyle/labels/good%20fourth%20issue
+    before taking up some bug fixes or features.
 
-Please follow the below links in order to start contributing to the checkstyle project
+## :book: Code of Conduct
 
-* [Beginning Development](https://checkstyle.org/beginning_development.html)
+This project and everyone participating in it is governed by the
+    [CheckStyle Code of Conduct](/.github/CODE_OF_CONDUCT.md).
 
-* [Contribution Guidelines](https://checkstyle.org/contributing.html)
+## :bulb: Asking Questions
 
-Find the links below to import the checkstyle project to an IDE
+See our [Discussions Page](https://github.com/checkstyle/checkstyle/discussions). In short, GitHub
+issues are not the appropriate place to debug your specific project, but should be reserved
+for filing bugs and feature requests.
 
-* [Importing checkstyle project to IntelliJ](https://checkstyle.org/idea.html)
+## :inbox_tray: Opening an Issue
 
-* [Importing checkstyle project to Eclipse](https://checkstyle.org/eclipse.html)
+Before [Opening an Issue](https://github.com/checkstyle/checkstyle/issues),
+check if you are using the latest version of the project,
+[found here](https://github.com/checkstyle/checkstyle/releases). If you are not up-to-date,
+see if updating to the latest release fixes your issue.
 
-* [Importing checkstyle project to Netbeans](https://checkstyle.org/netbeans.html)
+A great way to contribute to the project is to create a detailed issue when you encounter
+a problem or would like to suggest a feature. We always appreciate a well-written,
+thorough issue description. :brain:
+
+Some points to consider when opening an issue:
+- **Fully complete the provided issue template.** The bug report and feature request templates
+    specify all the information we need to quickly and efficiently address your issue. Be clear,
+    concise, and descriptive.
+    Provide as much information as you can, including steps to reproduce, stack traces, etc.
+- **Use [GitHub-flavored Markdown](https://help.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax).**
+    Especially put code blocks and console outputs in backticks (```). This improves readability.
+- **Do not open a duplicate feature request.** Search for existing feature requests first. If you
+    find your feature (or one very similar) previously requested, comment on that issue.
+
+## :lock: Reporting Security Issues
+
+**Do not file a public issue for security vulnerabilities.** Please contact the 
+maintainers directly.
+see the [Security Policy](https://github.com/checkstyle/SECURITY.md) for more information.
+
+## :technologist: Submitting Pull Requests
+
+ - **Read our pull request rules.** See [PR Rules](https://github.com/checkstyle/checkstyle/wiki/PR-rules).
+ - **Comment on the issue.** If you are working on an existing issue, please comment on the issue
+    to let others know that you are working on it ("I am on it.").
```

Comment:
I think this left open ended on what if there is no existing issue.

Also this point should first. PR rules was created for already opened PRs and should probably be last.

Maybe also make a point re pointing to https://checkstyle.org/contributing.html#Submitting_your_contribution and https://checkstyle.org/beginning_development.html#Starting_Development where we talk about remotes and branches? I think some work needs to be done with the external pages some. Things are a bit all over the place.

Reply:
>I think this left open ended on what if there is no existing issue.

fixed

>Maybe also make a point re pointing to

I would like to keep this as concise as possible and avoid repitition

Reply:
> fixed

I am sorry, I am not seeing how this one is fixed.